homeassistant/components/sensor/sense.py

```diff
@@ -150,3 +183,76 @@ def update(self):
             state = self._data.get_trend(self._sensor_type,
                                          self._is_production)
             self._state = round(state, 1)
+
+
+class SenseDevice(Entity):
+    """Implementation of a Sense detected device."""
+
+    def __init__(self, realtime_data_wrapper, name, location, icon):
+        """Initialize the sensor."""
+        self._realtime_data_wrapper = realtime_data_wrapper
+        self._device_name = name
+        if location != '':
+            self._device_name = '{} ({})'.format(name, location)
+        self._name = '{} Energy Usage'.format(self._device_name)
+        self._icon = icon
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def force_update(self):
+        """Return if an same sensor value should still count as an update."""
+        return True
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return 'W'
+
+    @property
+    def icon(self):
```

Comment:
I would create a dict with this mapping and save the if's tree

Reply:
Good call! I just pushed an update with a fix.